fmriprep/config.py

```diff
@@ -434,8 +440,12 @@ class execution(_Config):
     """Only build the reports, based on the reportlets found in a cached working directory."""
     run_uuid = f'{strftime("%Y%m%d-%H%M%S")}_{uuid4()}'
     """Unique identifier of this particular run."""
+    processing_groups = None
+    """List of tuples (participant, session(s)) that will be preprocessed."""
```

Comment:
This is derived from `participant_label`, `session_label` and `subject_anatomical_reference`. Might be handy for reporting, but it does mean that the config object state can be inconsistent.

IMO it would be better just to calculate the groups when needed with a well-isolated function rather than deal with (de)serialization and consistency.

Reply:
i also considered omitting, but decided to include it since having that provenance is nice. `session_label` is only populated if `--session-label` is used, otherwise all sessions pertaining to that subject are queried from the layout (and that information is not currently being stored). the config object should never be changed, just the dict representation of it (and when `dump`ing to toml).